sdk/cosmos/azure_data_cosmos/src/options/mod.rs

```diff
@@ -13,6 +15,32 @@ use std::fmt::Display;
 #[derive(Clone, Default, Debug)]
 pub struct CosmosClientOptions {
     pub client_options: ClientOptions,
+    pub application_name: Option<String>,
+    pub application_region: Option<String>,
+    pub application_preferred_regions: Option<Vec<String>>,
+    pub account_initialization_custom_endpoints: Option<HashSet<String>>,
+    pub consistency_level: Option<ConsistencyLevel>,
+    pub request_timeout: Duration,
+    pub enable_remote_region_preferred_for_session_retry: bool,
+    pub enable_partition_level_circuit_breaker: bool,
+    pub disable_partition_level_failover: bool,
+    pub enable_upgrade_consistency_to_local_quorum: bool,
+    pub throughput_bucket: Option<i32>,
+    pub session_retry_options: SessionRetryOptions,
```

Comment:
There's a general concern I have across several types in this PR: I'd really like to avoid adding stuff before we need it. This is a good opportunity to make sure we are seriously considering the _intent_ behind functionality in other SDKs. And it's very likely we'll decide to implement things in here in Rust in mostly the same way. However, I don't want to push us directly into copying the existing SDKs because we already copied all the data structures over.